pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -314,25 +298,18 @@ func getPoolByListenerID(client *gophercloud.ServiceClient, loadbalancerID strin
 			for _, l := range p.Listeners {
 				if l.ID == listenerID {
 					listenerPools = append(listenerPools, p)
+					break
 				}
 			}
 		}
-		if len(listenerPools) > 1 {
-			return false, ErrMultipleResults
-		}
 		return true, nil
 	})
 	if err != nil {
-		if isNotFound(err) {
-			return nil, ErrNotFound
-		}
 		return nil, err
 	}
 
 	if len(listenerPools) == 0 {
 		return nil, ErrNotFound
-	} else if len(listenerPools) > 1 {
-		return nil, ErrMultipleResults
```

Comment:
Code showed that if len(listenerPools) > 1 , ErrMultipleResults was returned before, but after your change, we will never return ErrMultipleResults